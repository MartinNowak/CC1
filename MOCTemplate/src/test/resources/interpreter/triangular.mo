DEF MAIN:nat == triangular(2000)

DEF triang2(sum:nat, idx:nat, n:nat):nat == IF or(lt(idx, n), eq(idx, n)) THEN triang2(add(sum, idx), add(idx, 1), n) ELSE sum FI

DEF triangular(n:nat):nat == triang2(0, 0, n)
