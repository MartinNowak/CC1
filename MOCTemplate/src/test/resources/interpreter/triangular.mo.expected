SUCCESS
2001000